crud/src/main/java/com/redhat/lightblue/crud/AbstractBulkJsonObject.java

```diff
@@ -0,0 +1,158 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.crud;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Collections;
+import java.util.Comparator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+
+import com.redhat.lightblue.util.JsonObject;
+import com.redhat.lightblue.Request;
+
+/**
+ * Base class for bulk request and responses. This class contains the
+ * code common to both, as bulk request and response are structurally
+ * similar. It deals with a JSON of the form:
+ *
+ *
+ * <pre>
+ *   {
+ *     "<entries>": [
+ *         {
+ *             "seq":0,
+ *             "op": "FIND",
+ *             "<entry>": { item }
+ *         }
+ *     ]
+ *   }
+ * </pre> 
+ */
+abstract class AbstractBulkJsonObject<T extends JsonObject> extends JsonObject {
+
+    protected final List<T> entries=new ArrayList<T>();
```

Comment:
No. Will change.

On Thu, Aug 27, 2015 at 11:37 AM, Dennis Crissman <notifications@github.com>
wrote:

> In
> crud/src/main/java/com/redhat/lightblue/crud/AbstractBulkJsonObject.java
> <https://github.com/lightblue-platform/lightblue-core/pull/437#discussion_r38123181>
> :
>
> > + *
> > + * <pre>
> > + *   {
> > + *     "<entries>": [
> > + *         {
> > + *             "seq":0,
> > + *             "op": "FIND",
> > + *             "<entry>": { item }
> > + *         }
> > + *     ]
> > + *   }
> > + * </pre>
> > + */
> > +abstract class AbstractBulkJsonObject<T extends JsonObject> extends JsonObject {
> > +
> > +    protected final List<T> entries=new ArrayList<T>();
>
> Does this need to be protected since getEntires() also exposes it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/437/files#r38123181>
> .
>